### PR TITLE
Add side-calendar to group form

### LINKED ISSUE #322: Help the user choose a non-conflicting start and end date
When someone visits the create group page not from the calendar page, they might not be aware of the selected floor's availability. To help them, and to avoid the need to open the calendar page separately, we can either validate the selected time intervals live (whenever it changes, show a warning that it conflicts), or maybe we can have a button that shows the currently selected room's calendar in a popup.

### PR DESCRIPTION
**Pull request by OmTheTurtle**
Closes #322

#### What's new

- Add a button that opens a sidenav with a calendar
- The calendar displays the selected floor's event in a day grid
- When the user changes the floor, the calendar refreshes with the correct data
- Separated `create` and `show` logic in client-side JS code for groups

### CODE REVIEW

**Comment by poiuztrewq1** [comment]
In dark-mode you can't properly see the text in the side-calendar. 
tanulo

File: public/js/group/create.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  event.preventDefault()
+  const formData = new FormData(event.target)
+  const errors = validateGroup(formData)
+
+  if (errors.length) {
+    clearMessages()
+    errors.forEach((err) => displayMessage('danger', err))
+  } else {
+    fetch('/groups', { method: 'POST', body: formData })
+      .then(async (res) => await handleResponse(res))
```
Comment (poiuztrewq1) [inline public/js/group/create.js:42]: I might be wrong, but I think you can simply pass async function in promise chain like: ` .then(handleResponse)`

File: views/group/new.pug [hunk 2]
```diff
@@ -49,21 +54,33 @@ block content
         button.button.is-link(type="submit", id="submitBtn", disabled)
           i.fas.fa-plus
           span &nbsp;Csoport létrehozása
+  #quickviewDefault.quickview
+    header.quickview-header
+      p.title Szint foglaltsága
+      span.delete(data-dismiss="quickview")
+    .quickview-body
+      .quickview-block
```
Comment (triszt4n) [inline views/group/new.pug:62]: Let's give `quickview-block` a bit of padding, e.g.: `.py-4.px-3`

File: public/js/group/create.js (hunk 1 above)
Comment (OmTheTurtle) [reply public/js/group/create.js:42]: You're not wrong, will refactor!

**Comment by OmTheTurtle** [comment]
Updated the PR according to comments

**Pull request merged**